controllers/ControllerComment.php

```diff
@@ -2,100 +2,100 @@
-	private object $_twig;
-	private object $_commentManager;
-	private string $_server;
-
-
-	public function __construct($method, $twig, $params){
-		$this->_method = $method;
-		$this->_twig = $twig;
-		$this->_params = $params;
-		$this->_commentManager = new CommentsManager();
-		$this->_server = ( new ConstantGlobal(ServerRequest::fromGlobals()) )->getServerName()['SERVER_NAME'];
-		$target = $method[2];
-		if (method_exists(ControllerComment::class, $target) ) {
-			$this->$target();
-		}else{
-			echo $this->_twig->render('404.twig');
-		}
-	}
-
-	public function addCommentPost(){
+	/**
+	 *Ajoute un commentaire à post en BDD.
+	 *
+	 * Si l'utilisateur n'est pas connecté, renvoi sur la page de connexion.
+	 */
+	public function addCommentPost(): void {
 		$key = (new Session)->getKey('user');
 
 
 		if (!isset($key)){
-			header("Location: http://".$this->_server."/Auth/login");
+			header("Location: http://".$this->getServer()."/Auth/login");
 			exit();
 		}else{
 
 			$userId = $key['id'];
-			$comment = htmlentities(trim($this->_params['message']));
+			$comment = htmlentities(trim($this->getPost()['message']));
 			$postId = (new Session)->getKey('post');
 
-			$good = $this->_commentManager->addComment($userId, $comment, $postId['id']);
+			$good = (new CommentsManager())->addComment($userId, $comment, $postId['id']);
 
 			if ($good){
-				header("Location: http://".$this->_server."/Post/".$postId['id']);
+				header("Location: http://".$this->getServer()."/Post/viewPost/".$postId['id']);
 			}else{
 				echo 'Une erreur c\'est produite veuillez recommencer !' ;
 			}
 		}
 	}
 
-	private function updateComment(){
+	/**
+	 * Modification d'un commentaire en BDD.
+	 *
+	 * Renvoi sur la page du post concerné.
+	 */
+	public function updateComment(): void {
 
-		$key = (new Session)->getKey('post');
-		$commentId = $this->_method[3];
-		$post = $this->_params;
+		$keyPost = (new Session)->getKey('post');
+		$keyUser = (new Session)->getKey('user');
+		$post = $this->getPost();
 
 		if (isset($post['message'])){
-			$this->checkComment($post['message'], $commentId, $key['id']);
+			$this->checkComment($post['message'], $this->getSlug(), $keyPost['id']);
 		}
 
+		$content = (new CommentsManager())->oneComment((int) $this->getSlug());
 
-
-		$content = $this->_commentManager->oneComment((int) $commentId);
-
-		echo $this->_twig->render( 'comment.twig', ['logged'=> TRUE,
-		                                            'server' => $this->_server,
-		                                            'id' => $content['id'],
-		                                            'content'=> html_entity_decode($content['content'])]);
+		echo $this->getTwig()->render( 'comment.twig', ['logged'=> TRUE,
+		                                                'server' => $this->getServer(),
+		                                                'id' => $content['id'],
+		                                                'admin' => $keyUser['admin'],
```

Comment:
Codacy found an issue: [All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$keyUser['admin']'.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=6061935)